model-optimizer/mo/graph/connection.py

```diff
@@ -174,6 +233,19 @@ def check_and_remove_edge():
                 data_node = self.source._create_data_if_necessary()
                 edge_attrs = check_and_remove_edge() or {}
                 edge_attrs.update({'in': port.idx})
+
+                if port.idx in port.node.in_nodes():
+                    dest_attrs = port.node.in_node(port.idx).attrs()
+                    fw_name_attr = "fw_tensor_debug_info"
+                    if fw_name_attr in dest_attrs:
+                        if attributes_save_mode == "merge":
+                            if fw_name_attr in data_node:
+                                data_node[fw_name_attr] += dest_attrs[fw_name_attr]
+                            else:
+                                data_node[fw_name_attr] = dest_attrs[fw_name_attr]
+                        elif attributes_save_mode == "dest":
+                            data_node[fw_name_attr] = dest_attrs[fw_name_attr]
```

Comment:
These lines almost equal to lines in 141-152. Consider refactoring code and create a function.